index.bs

```diff
@@ -1129,13 +1138,61 @@ a numbered step. If outdented, it (today) is rendered as a bullet in the midst o
+
+                  1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                      <dl class="switch">
+                          :   is set to {{ResidentKeyRequirement/required}}
+                          ::  If the |authenticator| is not capable of storing a [=client-side-resident public key credential
+                              source=], [=iteration/continue=].
+
+                          :   is set to {{ResidentKeyRequirement/preferred}} or {{ResidentKeyRequirement/discouraged}}
+                          ::  No effect.
+
+                          :   is [=not present=]
+                          ::  if <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code>
+                              is set to [TRUE] and the |authenticator| is not capable of storing a [=client-side-resident public
+                              key credential source=], [=iteration/continue=].
+                      </dl>
+
                   1. If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{AuthenticatorSelectionCriteria/userVerification}}</code> is
                     set to {{UserVerificationRequirement/required}} and the |authenticator| is not capable of performing [=user
                     verification=], [=iteration/continue=].
 
+            1. Let |requireResidentKey| be the <dfn>effective resident key requirement for credential creation</dfn>, a Boolean value,
+                as follows. 
+
+                If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{residentKey}}</code>
+
+                    <dl class="switch">
+
+                        :   is present and set to {{ResidentKeyRequirement/discouraged}}
+                        ::  Let |requireResidentKey| be [FALSE].
+
+                        :   is present and set to {{ResidentKeyRequirement/preferred}}
+                        ::  If the |authenticator|
+
+                            <dl class="switch">
+                                :   is capable of [=client-side-resident public key credential source=]
+                                ::  Let |requireResidentKey| be [TRUE].
+
+                                :   is not capable of [=client-side-resident public key credential source=], or if the client cannot determine authenticator capability,
+                                ::  Let |requireResidentKey| be [FALSE].
+                            </dl>
+
+                        :   is present and set to {{ResidentKeyRequirement/required}}
+                        ::  Let |requireResidentKey| be [TRUE].
+
+                        :   is [=not present=]
+                        ::  If <code>|options|.{{PublicKeyCredentialCreationOptions/authenticatorSelection}}.{{requireResidentKey}}</code>
+                                is [=present=], let |requireResidentKey| be the value of
```

Comment:
nit: `requireResidentKey` has a default value and thus I believe is always considered to be present. So here you could just take the value of `requireResidentKey`.

Reply:
Agreed - simplified based on this assertion.